### ISSUE TITLE
`with_fullscreen`/`set_fullscreen` do not fullscreen

### ISSUE THREAD

**Issue by lylythechosenone** [rust-windowing/winit#3628]
### Description

Using `WindowBuilder::with_fullscreen` results in a maximized window (see video), and `Window::set_fullscreen` does nothing at all.

I am using borderless fullscreen without a monitor set. Exclusive does work as intended, video mode and all.

https://github.com/rust-windowing/winit/assets/19551513/37c37434-6e83-4f47-a640-0b93f5ce6458

### macOS version

```shell
ProductName:		macOS
ProductVersion:		14.0
BuildVersion:		23A344
```

### Winit version

0.29.14

**Comment by madsmtm** [comment]
Create the window inside of your event loop in the `Event::NewEvents(StartCause::Init)` event, like this:
```rust
let mut window = None;
event_loop.run(|event, event_loop| {
    match event {
        Event::NewEvents(StartCause::Init) => {
            window = Some(
                WindowBuilder::new()
                // ...
            );
        }
        _ => {}
    }
})
```

**Comment by lylythechosenone** [comment]
Oh, I see. Maybe this should go in the docs, then.

Thanks for your help.

**Comment by madsmtm** [comment]
It's been fixed in v0.30, which we're releasing in a few days. It was stated in [the README before](https://crates.io/crates/winit/0.29.15).
